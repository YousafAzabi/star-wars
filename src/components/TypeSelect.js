import React from 'react';

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  formControl: {
    float: 'left',
    width: '34%',
    minWidth: 200,
    marginLeft: '8%',
    marginRight: '8%'
  },
});

class TypeSelect extends React.Component {

  render() {
    const { classes, handleChange } = this.props;
    
    return (
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="demo-controlled-open-select">Collection</InputLabel>
        <Select value={this.props.type}
        onChange={handleChange}>
          <MenuItem value={'people'}>People</MenuItem>
          <MenuItem value={'starships'}>StarShips</MenuItem>
        </Select>
      </FormControl>
    );
  }
}

export default withStyles(styles)(TypeSelect);
