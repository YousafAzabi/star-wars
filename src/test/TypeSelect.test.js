import React from "react";
import { create } from "react-test-renderer";
import TypeSelect from "../components/TypeSelect";

describe("TypeSelect component", () => {
  test("it matches the snapshot", () => {
    const component = create(<TypeSelect type='people'
      handleChange={console.log('change')} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("it matches the snapshot", () => {
    const component = create(<TypeSelect type='starships'
      handleChange={console.log('change')} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("test display value equals 'starships'", () => {
    const component = create(<TypeSelect type='starships'
      handleChange={console.log('change')} />);
    const rootInstance = component.root;
    const select = rootInstance.findByType('input');
    expect(select.props.value).toBe('starships');
  });
});
