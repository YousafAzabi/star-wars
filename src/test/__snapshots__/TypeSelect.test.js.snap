// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeSelect component it matches the snapshot 1`] = `
<div
  className="MuiFormControl-root TypeSelect-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="demo-controlled-open-select"
  >
    Collection
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="true"
      aria-pressed="false"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      People
    </div>
    <input
      type="hidden"
      value="people"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`TypeSelect component it matches the snapshot 2`] = `
<div
  className="MuiFormControl-root TypeSelect-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="demo-controlled-open-select"
  >
    Collection
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="true"
      aria-pressed="false"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      StarShips
    </div>
    <input
      type="hidden"
      value="starships"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
